ColabTool/Shared/PostDetails/index.jsx

```diff
@@ -80,94 +82,93 @@ const renderControls = ({
       <Text size={'small'}> Deleting... </Text>
     );
   }
+
+  if (manager && isWorking && !isDeleting) {
+    return (
+      <Text size={'small'}> Approving... </Text>
+    );
+  }
+
   return (
     <div>
-      {renderEdit({
-        onEditClick,
-      })}
       {renderDelete({
         isConfirmingDelete,
         onCancelConfirmClick,
         onDeleteConfirmClick,
         onDeleteClick,
       })}
+      {renderEdit({
+        onEditClick,
+      })}
+      {manager && renderApproval({
+        onApproveClick,
+      })}
     </div>
   );
 };
 
-const renderPostTypeIcon = (postType) => {
-  switch (postType) {
-    case 'image':
-      return (<ImageIcon size={'small'} />);
-    case 'retweet':
-      return (<RetweetIcon size={'small'} />);
-    default:
-      return (<LinkIcon size={'small'} />);
-  }
-};
-
 /* eslint-enable react/prop-types */
 
 const PostDetails = ({
   isDeleting,
   isConfirmingDelete,
+  isWorking,
+  manager,
+  onApproveClick,
   onCancelConfirmClick,
   onDeleteClick,
   onDeleteConfirmClick,
   onEditClick,
-  profile,
-  postType,
+  draftDetails,
 }) =>
   <div className={style['post-details']}>
+    <div className={style['post-action-details']}>
+      <span className={style['post-icon']}>
```

Comment:
Heya @stevemdixon - oh good one 😄  So I removed the span and added the styling directly to the icon itself. It _almost_ works but the icon seems to sit a bit below the text:
<img width="660" alt="screen shot 2017-03-13 at 11 13 35 am" src="https://cloud.githubusercontent.com/assets/6446462/23860778/4a850368-07de-11e7-92f4-dd985a6ed102.png">

My hunch is that we want to keep it in line with the text. Curious to hear what you think. 😄